app/controllers/users_controller.rb

```diff
@@ -21,81 +21,83 @@ def update_assignments
+          rescue Octokit::Error => e
+              Rails.logger.error "Failed to update GitHub permissions: #{e.message}"
+              flash[:alert] = "Failed to update assignments. Please try again."
+              @assignments = Assignment.all
+              render :show
+          end
         else
-            @assignments ||= []
-            flash.now[:alert] = "Failed to update assignments. Please try again."
-            render :show
+          @assignments ||= []
+          flash.now[:alert] = "Failed to update assignments. Please try again."
+          render :show
         end
-    end
+      end
 
-      private
+      def update_permissions(assignment_ids, role)
+        assignment_ids.each do |assignment_id|
+          permission = Permission.find_or_initialize_by(user: @user, assignment_id: assignment_id)
+          permission.update(role: role)
+        end
+      end
 
       def update_github_permissions(user)
         access_token = session[:github_token]
         client = Octokit::Client.new(access_token: access_token)
         org_name = "AutograderFrontend"
-
-        # All assignments that need to be updated
         @assignments = Assignment.all
 
-
         @assignments.each do |assignment|
           repo_identifier = "#{org_name}/#{assignment.repository_name}"
-
-          # Fetch the user's permission for the current assignment
           permission = Permission.find_by(user: user, assignment: assignment)
 
+          update_collaborator_permissions(client, repo_identifier, user, permission, assignment)
+        end
+      end
 
-          if permission.role == "no_permission" || permission.role.nil?
-            begin
-              client.remove_collaborator(repo_identifier, user.name)
-              Rails.logger.info "Removed collaborator #{user.name} from #{repo_identifier} due to 'none' permission"
-            rescue Octokit::Error => e
-              Rails.logger.error "Failed to remove collaborator #{user.name} from #{repo_identifier}: #{e.message}"
-              raise
-            end
-          elsif permission.role == "read_write" || permission.role == "read"
-            # Use ternary operator for selecting permission
-            github_permission = (permission.role == "read_write") ? "push" : "pull"
-            begin
-              client.add_collaborator(repo_identifier, user.name, permission: github_permission)
-              Rails.logger.info "Updated collaborator #{user.name} on #{repo_identifier} with #{github_permission} access"
-            rescue Octokit::Error => e
-              Rails.logger.error "Failed to update collaborator #{user.name} on #{repo_identifier}: #{e.message}"
-              raise
-            end
-          else
-            Rails.logger.error "Unknown permission role: #{permission.role} for user #{user.name} on assignment #{assignment.id}"
-          end
+      private
+
+      def update_collaborator_permissions(client, repo_identifier, user, permission, assignment)
```

Comment:
Same comment here as I made previously for `update_collaborator_permissions` method in `assignments_controller.rb` file.

Please see if providing the `assignment` argument can be avoided somehow.

Reply:
Got it.